zulip/zulip/__init__.py

```diff
@@ -753,16 +744,23 @@ def do_register() -> Tuple[str, int]:
             if queue_id is None:
                 (queue_id, last_event_id) = do_register()
 
-            res = self.get_events(queue_id=queue_id, last_event_id=last_event_id)
+            try:
+                res = self.get_events(queue_id=queue_id, last_event_id=last_event_id)
+            except (
+                requests.exceptions.Timeout,
+                requests.exceptions.SSLError,
+                requests.exceptions.ConnectionError,
+            ):
+                if self.verbose:
+                    print(f"Connection error:\n{traceback.format_exc()}")
+            except Exception:
+                if self.verbose:
+                    print(f"Unexpected error:\n{traceback.format_exc()}")
```

Comment:
I tested this, and these needed a `sleep` + `continue` to avoid an error with `res` being an undefined variable when falling through here.  I'll merge after applying that change.